pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/InvertedIndexReader.java

```diff
@@ -24,13 +24,14 @@
 public interface InvertedIndexReader<T> extends Closeable {
 
   /**
-   * Get the document ids for the given dictionary id.
-   * @param dictId dictionary ID
+   * Returns the document ids for the given dictionary id.
    */
   T getDocIds(int dictId);
 
   /**
-   * Get the document ids for a given value
+   * Returns the document ids for the given string representation of the value.
```

Comment:
Can we add APIs for each raw value type so that this interface can also be used for non dictionary based inverted indexes? With the current change, we can probably still do but the raw value has to be converted to string?

Reply:
The value here is from the query, so we always pass it in as string, and the inverted index implementation can parse the string based on the value type. This is similar to how dictionary handles the lookup of user input value (`Dictionary.indexOf(String stringValue)`).